reframe/core/runtime.py

```diff
@@ -0,0 +1,252 @@
+#
+# Handling of the current host context
+#
+
+import os
+import functools
+import re
+import shutil
+import socket
+from datetime import datetime
+
+import reframe.core.config as config
+import reframe.core.fields as fields
+import reframe.utility.os_ext as os_ext
+from reframe.core.exceptions import (ConfigError,
+                                     ReframeFatalError,
+                                     SpawnedProcessError,
+                                     UnknownSystemError)
+from reframe.core.modules import ModulesSystem
+
+
+class HostSystem:
+    def __init__(self, system, partname=None):
+        self._system = system
+        self._partname = partname
+
+    def __getattr__(self, attr):
+        # Delegate any failed attribute lookup to our backend
+        return getattr(self._system, attr)
+
+    # We redefine the following properties just to document them for the users
+    @property
+    def name(self):
+        """The name of this system."""
+        return self._system.name
+
+    @property
+    def descr(self):
+        """The description of this system."""
+        return self._system.descr
+
+    @property
+    def resourcesdir(self):
+        """Global resources directory for this system.
+
+        You may use this directory for storing large resource files of your
+        regression tests.
+        See `here <configure.html#system-configuration>`__ on how to configure
+        this.
+
+        :type: :class:`str`
+        """
+        return self._system.resourcesdir
+
+    @property
+    def partitions(self):
+        """Get all the partitions of this system.
```

Comment:
In agreement with the way you did the comments for `name`, `descr` and `resourcesdir`, I would note here instead:
```
"""All the partitions of this system.
```
or
```
"""The partitions of this system.
```